pkgs/development/haskell-modules/lib/compose.nix

```
@@ -285,7 +285,9 @@ rec {
     enableLibraryProfiling = false;
     isLibrary = false;
     doHaddock = false;
-    postFixup = "rm -rf $out/lib $out/nix-support $out/share/doc";
+    postFixup = drv.postFixup or "" + ''
+      rm -rf $out/lib $out/nix-support $out/share/doc
+    '';
```

This ensures it doesn't break even if previous `postFixup` isn't ending in a newline.

```suggestion
    postFixup = ''
      rm -rf $out/lib $out/nix-support $out/share/doc
    '' + drv.postFixup or "";
```

---

We can’t put it after because it would invert the order in which the script is concatenated. We could put an extra empty newline before.

---

But it didn’t have a leading newline before, so nothing should break by that change … oh wait it just threw away the stuff :)

---

I guess the intuition would be the step would be added after, yes.

---

fixed ptal

---

I think this wouldn't work because the bash parser only parses a hash as the start of a comment if there's a space-like thing before.

---

Added another empty line before, ptal